### PR TITLE
006-1 Foundation Guardian Changes 

### PR DESCRIPTION
**Pull request by maurelian**
1. Reinitializes the SuperchainConfig with the Security Council as Guardian
1. Transfers ownership of the ProxyAdmin to a new 2 of 2 safe

Stacked on #165.

The SignFromJSON script is intended to have a minimal diff compared to the preceding guardian runbook, so diffing can be helpful for reviewing:

```
diff  006-0-sc-guardian-updates/SignFromJson.s.sol 006-1-foundation-guardian-updates/SignFromJson.s.sol
```

### CODE REVIEW

File: tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+        shouldHaveCodeExceptions[5] = batchInboxAddress;
+
+        for (uint256 i = 0; i < securityCouncilSafeOwners.length; i++) {
+            shouldHaveCodeExceptions[6 + i] = securityCouncilSafeOwners[i];
+        }
+
+        return shouldHaveCodeExceptions;
+    }
+
+    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/extra/addresses/mainnet/op.json
```
Comment (blmalone) [inline tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol:501]:
```suggestion
    /// @notice Reads the contract addresses from lib/superchain-registry/superchain/extra/addresses/sepolia/op.json
```

File: tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+    using LibString for string;
+
+    // Chains for this task.
+    string constant l1ChainName = "sepolia";
+    string constant l2ChainName = "op";
+
+    // Safe contract for this task.
+    GnosisSafe securityCouncilSafe = GnosisSafe(payable(0xa87675ebb9501C7baE8570a431c108C1577478Fa));
+    GnosisSafe foundationSafe = GnosisSafe(payable(0xDEe57160aAfCF04c34C887B5962D0a69676d3C8B));
+    GnosisSafe upgradeOwnerSafe = GnosisSafe(payable(0xeD3d7D9f610a8ACcBe9CACA172B7F3d70530E89D));
```
Comment (blmalone) [inline tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol:77]: How can I verify this?

File: tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol [hunk 3]
```diff
@@ -0,0 +1,537 @@
+// SPDX-License-Identifier: MIT
```
Comment (blmalone) [inline tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol:1]: Screenshot 2024-04-19 at 10 27 05
Getting an error when I simulate.

**Review by blmalone — COMMENTED** [review]
Left some comments here for now, haven't looked at the `VALIDATION.md` file yet.

File: tasks/sep/006-1-foundation-guardian-updates/VALIDATION.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+- **Key:** 0xd30e835d3f35624761057ff5b27d558f97bd5be034621e62240e5c0b784abe68
+  **Before:** 0x000000000000000000000000dee57160aafcf04c34c887b5962d0a69676d3c8b
+  **After:** 0x000000000000000000000000a87675ebb9501c7bae8570a431c108c1577478fa
+  **Meaning:** The Guardian address has been updated from `0xdee57160aafcf04c34c887b5962d0a69676d3c8b` (Foundation) to `0xa87675ebb9501c7bae8570a431c108c1577478fa` (Council).
+    The key is `keccak256("superchainConfig.guardian") - 1` ([ref](https://github.com/ethereum-optimism/optimism/blob/maur/sepolia-council/packages/contracts-bedrock/src/L1/SuperchainConfig.sol#L23)),
+    which can be verified using `cast keccak "superchainConfig.guardian"`.
+
+### `0xdee57160aafcf04c34c887b5962d0a69676d3c8b` (`GnosisSafe`)
+
+The nonce is increased.
```
Comment (blmalone) [inline tasks/sep/006-1-foundation-guardian-updates/VALIDATION.md:53]:
```suggestion
State Changes:

- **Key:** 0x0000000000000000000000000000000000000000000000000000000000000005
  **Before:** 0x000000000000000000000000000000000000000000000000000000000000000a
  **After:** 0x000000000000000000000000000000000000000000000000000000000000000b
The nonce is increased from `0xa` (decimal 10) to `0xb` (decimal 11).
```

Comment (blmalone) [reply tasks/sep/006-1-foundation-guardian-updates/VALIDATION.md:53]: The last line here is taken from: https://github.com/ethereum-optimism/superchain-ops/blob/f3f4285f86449dc4c03fc3cf1643cab70481997a/tasks/eth/metal-001-MCP-L1/VALIDATION.md?plain=1#L339-L341 

You might want to add more onto it.

File: tasks/sep/006-1-foundation-guardian-updates/input.json [hunk 5]
```diff
@@ -0,0 +1,88 @@
+{
```
Comment (blmalone) [inline tasks/sep/006-1-foundation-guardian-updates/input.json:1]: Same question as the other PR: https://github.com/ethereum-optimism/superchain-ops/pull/165/files#r1572450219

Comment (maurelian) [reply tasks/sep/006-1-foundation-guardian-updates/input.json:1]: Same response: validation of changes is sufficient to validate generation. :)

File: tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol (hunk 3 above)
Comment (maurelian) [reply tasks/sep/006-1-foundation-guardian-updates/SignFromJson.s.sol:1]: yep, that's intentional.
What's happening is that this runbook assumes that the previous one has already been executed. Once it has, those errors will disappear.

File: tasks/sep/006-1-foundation-guardian-updates/input.json [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        "inputs": [
+          { "name": "newOwner", "type": "address", "internalType": "address" }
+        ],
+        "name": "transferOwnership",
+        "payable": false
+      },
+      "contractInputsValues": {
+        "newOwner": "0xeD3d7D9f610a8ACcBe9CACA172B7F3d70530E89D"
+      }
+    }
```
Comment (maurelian) [inline tasks/sep/006-1-foundation-guardian-updates/input.json:86]: Remove this for just the Guardian update.

**Pull request closed without merging**